### DISCUSSION TITLE
How to publish self-contained WinUI3 app to Microsoft store?

### DISCUSSION THREAD

**Discussion by Odinodin** [microsoft/WindowsAppSDK#3295] [Q&A] [answered]
Is it possible to publish a self-contained WinUI3 app to the Microsoft store? 

By self-contained I mean an app which bundles the .NET runtime. Currently, the app we're making prompts the users to install the runtime when installing the app from the Microsoft Store.

We're using Visual Studio 2022 "Package and publish > Create App Packages > Microsoft Store" wizard to generate the package.
The WindowsAppSdk version is 1.2.221209.1.

Here's an excerpt from the csproj-file in the project.
```
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <OutputType>WinExe</OutputType>
	<TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
	<TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
	<RootNamespace>Example</RootNamespace>
	<ApplicationManifest>app.manifest</ApplicationManifest>
	<Platforms>x86;x64;arm64</Platforms>
	<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
	<PublishProfile>win10-$(Platform).pubxml</PublishProfile>
	<UseWinUI>true</UseWinUI>
	<EnableMsixTooling>true</EnableMsixTooling>
	<GenerateAppInstallerFile>False</GenerateAppInstallerFile>
	<AppxAutoIncrementPackageRevision>true</AppxAutoIncrementPackageRevision>
	<AppxSymbolPackageEnabled>true</AppxSymbolPackageEnabled>
	<AppxBundle>Always</AppxBundle>

        <!-- Changing this property does not change the output MSIX file size -->
	<SelfContained>true</SelfContained>
        <!-- This property works fine, the file size increases when it is set to true -->
	<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>
	<WindowsPackageType>MSIX</WindowsPackageType>
	<AppxBundlePlatforms>x64</AppxBundlePlatforms>
    </PropertyGroup>
    ...
</Project>
```

**Comment by AdamBraden** [comment, marked as answer]
@mikebattista ?

**Reply by mikebattista** [reply]
Is the app fully self-contained when run locally? Is it only after running the Package and Publish wizard that the resulting MSIX is framework-dependent on .NET?

**Reply by Odinodin** [reply]
I figured it out through [this official sample](https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/SelfContainedDeployment), seems to work fine now. The trick was to reference a designated `.pubxml` file from the `.csproj` file with `SelfContained` set. Setting `SelfContained` inside the `.csproj` file had no effect.

**Comment by carterlist** [comment]
A far easier approach is to set the self-contained property in your application project-
<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>

This way the MSIX installer project will build a self-contained package.
